### PR TITLE
add created_at to dataset and Campaign

### PR DESCRIPTION
**Pull request by Qouagga**
#80

### CODE REVIEW

File: backend/api/migrations/0023_auto_20231020_1755.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        ),
+        migrations.AddField(
+            model_name="annotationcampaign",
+            name="created_at",
+            field=models.DateTimeField(blank=True, null=True),
+        ),
+        migrations.AddField(
+            model_name="dataset",
+            name="created_at",
+            field=models.DateTimeField(blank=True, null=True),
```
Comment (rvovard) [inline backend/api/migrations/0023_auto_20231020_1755.py:29]: I think there's something wrong with this migration.
For this PR, we can see that GitHub actions for backend fail on `if [ "$(poetry run ./manage.py makemigrations --dry-run)" != "No changes detected" ]; then exit 1; fi`

When I try to run `poetry run ./manage.py makemigrations --dry-run` there is indeed a migration to be created. My guess is that current migration has been created before some changes in the PR.

I'm not familiar with Django migrations, so I can't help much. But if I had to repair it, here is what I would have tried:

* Empty database
* Delete migration file `0023_auto_20231020_1755.py`
* Migrate (`poetry run ./manage.py migrate`)
* Recreation migration (`poetry run ./manage.py makemigrations`)

**Review by rvovard — CHANGES_REQUESTED** [review]
Front code is OK, but it seems that something's wrong with the backend migrations.

File: backend/api/serializers/annotation_campaign.py [hunk 2]
```diff
@@ -192,6 +194,7 @@ def create(self, validated_data):
             annotation_scope=validated_data["annotation_scope"],
             owner_id=validated_data["owner_id"],
             instructions_url=validated_data.get("instructions_url"),
+            created_at=datetime.today(),
```
Comment (ixio) [inline backend/api/serializers/annotation_campaign.py:197]: This is no longer needed, right ?

File: pyproject.toml [hunk 3]
```diff
@@ -13,6 +13,7 @@ gunicorn = "^20.1.0"
 psycopg2-binary = "^2.8.6"
 sentry-sdk = "^1.3.1"
 drf-spectacular = "^0.26.0"
+freezegun = "^1.2.2"
```
Comment (ixio) [inline pyproject.toml:16]: This should be a dev.dependencies, it's only for testing not for production :thinking:

File: backend/api/migrations/0023_auto_20231020_1755.py (hunk 1 above)
Comment (ixio) [reply backend/api/migrations/0023_auto_20231020_1755.py]: @Qouagga either follow or at minimal you should rebase on master and rename to `0024_add_created_at_fields.py` (+ update the dependencies) :wink:

Also you'll need to fix the poetry.lock conflict (ask me if you have any difficulties with this).

**Review by ixio — APPROVED** [review]
LGTM

**Pull request merged**